### PR TITLE
Changes to compile for morty branches

### PR DESCRIPTION
**Pull request by HaseenaSainul**
1. bbclass named : bluetooth and features_check added to get these class support in morty version
2. cmake 3.7 recipe added to ensure minimum cmake available in morty is compatible with Thunder requirement
3. opkg patch updated to avoid warning during the time of applying patches
4. lib cobalt recipe updated to apply GCC patches only based on GCC version
5. libepoxy_1.4.0, openjpeg_2.3.1 recipe added to get wpewebkit expecting versions in morty build
6. GStreamer-1.10.4 used to ensure minimum available morty is compatible for wpewebkit requirements
7. WPEWebkit: 
    a. GStreamer OpenGL and MPG123 plugin selection added based on DISTRO_CODENAME to get the selection from corresponding providers
   b. woff2 selection added based on DISTRO_CODENAME
8. d.getVar implicit field added to avoid format check error from Morty version
9. DEPENDS += and SRC_URI+=  changed to _append
10. Some minor formatting changes

### CODE REVIEW

File: recipes-multimedia/gstreamer/gstreamer1.0-omx.inc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+SUMMARY = "OpenMAX IL plugins for GStreamer"
+HOMEPAGE = "http://gstreamer.freedesktop.org/"
+SECTION = "multimedia"
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-omx.inc:3]: Could you please add DESCRIPTION?

File: recipes-multimedia/gstreamer/gstreamer1.0-omx.inc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+LICENSE = "LGPLv2.1"
+LICENSE_FLAGS = "commercial"
+
+DEPENDS = "gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-bad"
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-omx.inc:8]: Could you please use DEPENDS_append here?

File: recipes-multimedia/gstreamer/gstreamer1.0-omx_1.10.4.bb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+include gstreamer1.0-omx.inc
+
+LIC_FILES_CHKSUM = "file://COPYING;md5=4fbd65380cdd255951079008b364516c \
+                    file://omx/gstomx.h;beginline=1;endline=21;md5=5c8e1fca32704488e76d2ba9ddfa935f"
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-omx_1.10.4.bb:4]: Could  you please align those lines with 4 spaces?

File: recipes-multimedia/gstreamer/gstreamer1.0-omx_1.10.4.bb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+SRC_URI = "http://gstreamer.freedesktop.org/src/gst-omx/gst-omx-${PV}.tar.xz \
+		  file://0001-Don-t-try-to-acquire-buffer-when-src-pad-isn-t-activ.patch \
+		  file://0002-fix-decoder-flushing.patch \
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-omx_1.10.4.bb:8]: Could  you please align those lines with 4 spaces?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_1.10.4.bb [hunk 5]
```diff
@@ -0,0 +1,36 @@
+require gstreamer1.0-plugins-bad.inc
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_1.10.4.bb:1]: Could  you please align this file with 4 spaces?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-base.inc [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    --enable-zlib \
+    --enable-introspection=no \
+"
+
+CACHED_CONFIGUREVARS_append_x86 = " ac_cv_header_emmintrin_h=no ac_cv_header_xmmintrin_h=no"
+
+FILES_${MLPREFIX}libgsttag-1.0 += "${datadir}/gst-plugins-base/1.0/license-translations.dict"
+
+do_compile_prepend() {
+        export GIR_EXTRA_LIBS_PATH="${B}/gst-libs/gst/tag/.libs:${B}/gst-libs/gst/video/.libs:${B}/gst-libs/gst/audio/.libs:${B}/gst-libs/gst/rtp/.libs"
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-base.inc:51]: Could you please indent it to 4 spaces?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-base_1.10.4.bb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+require gstreamer1.0-plugins-base.inc
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-base_1.10.4.bb:1]: Please indent file with 4 spaces

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-base.inc [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    ${@bb.utils.contains('DISTRO_FEATURES', 'alsa', 'alsa', '', d)} \
+    ${@bb.utils.contains('DISTRO_FEATURES', 'x11', 'x11', '', d)} \
+    gio-unix-2.0 ogg pango opus theora vorbis \
+"
+
+X11DEPENDS = "virtual/libx11 libsm libxrender libxv"
+X11ENABLEOPTS = "--enable-x --enable-xvideo --enable-xshm"
+X11DISABLEOPTS = "--disable-x --disable-xvideo --disable-xshm"
+
+PACKAGECONFIG[alsa]         = "--enable-alsa,--disable-alsa,alsa-lib"
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-base.inc:29]: Could you please align the spaces to match the agreed style?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good.inc [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+PACKAGECONFIG ??= " \
+    ${GSTREAMER_ORC} \
+    ${@bb.utils.contains('DISTRO_FEATURES', 'pulseaudio', 'pulseaudio', '', d)} \
+    ${@bb.utils.contains('DISTRO_FEATURES', 'x11', 'x11', '', d)} \
+    cairo flac gdk-pixbuf gudev jpeg id3demux icydemux libpng matroska soup speex taglib udp v4l2 \
+"
+
+X11DEPENDS = "virtual/libx11 libsm libxrender libxfixes libxdamage"
+
+PACKAGECONFIG[cairo]      = "--enable-cairo,--disable-cairo,cairo"
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-good.inc:18]: Could you please get rid off additional spaces?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.10.4.bb [hunk 10]
```diff
@@ -0,0 +1,35 @@
+require gstreamer1.0-plugins-good.inc
+
+LIC_FILES_CHKSUM = "file://COPYING;md5=a6f89e2100d9b6cdffcea4f398e37343 \
+                    file://common/coverage/coverage-report.pl;beginline=2;endline=17;md5=a4e1830fce078028c8f0974161272607 \
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.10.4.bb:4]: Could you please allign?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.16.%.bbappend [hunk 11]
```diff
@@ -2,7 +2,7 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/${BPN}:"
 
 PACKAGECONFIG_remove = "gtk"
 
-SRC_URI += "\
+SRC_URI_append = "\
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.16.%.bbappend:5]: Could you please add a space between " and \ ?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-ugly.inc [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+inherit gettext
+
+PACKAGECONFIG ??= " \
+    ${GSTREAMER_ORC} \
+    a52dec lame mpg123 mpeg2dec \
+"
+
+PACKAGECONFIG_remove = "a52dec lame mad mpeg2dec"
+
+PACKAGECONFIG[a52dec]   = "--enable-a52dec,--disable-a52dec,liba52"
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-ugly.inc:17]: Can we please get rid of additional spaces?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-ugly.inc [hunk 13]
```diff
@@ -0,0 +1,34 @@
+require gstreamer1.0-plugins.inc
+
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-ugly.inc:2]: Could you please add a DESCRIPTION?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-ugly_1.10.4.bb [hunk 14]
```diff
@@ -0,0 +1,13 @@
+require gstreamer1.0-plugins-ugly.inc
+
+LIC_FILES_CHKSUM = "file://COPYING;md5=a6f89e2100d9b6cdffcea4f398e37343 \
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-ugly_1.10.4.bb:3]: Could you please indent it with 4 spaces?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc [hunk 15]
```diff
@@ -0,0 +1,51 @@
+SUMMARY = "Plugins for the GStreamer multimedia framework 1.x"
+HOMEPAGE = "http://gstreamer.freedesktop.org/"
+BUGTRACKER = "https://bugzilla.gnome.org/enter_bug.cgi?product=Gstreamer"
+SECTION = "multimedia"
+
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc:5]: Could you please add a DESCRIPTION?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+DEPENDS = "gstreamer1.0 glib-2.0-native"
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc:6]: Could you please use _append here?

File: recipes-multimedia/gstreamer/gstreamer1.0_1.10.4.bb [hunk 17]
```diff
@@ -0,0 +1,18 @@
+require gstreamer1.0.inc
+
+LIC_FILES_CHKSUM = "file://COPYING;md5=6762ed442b3822387a51c92d928ead0d \
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0_1.10.4.bb:3]: Could you please indent it with 4 spaces?

File: recipes-multimedia/gstreamer/gstreamer1.0_1.16.%.bbappend [hunk 18]
```diff
@@ -1,5 +1,7 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/${BPN}:"
 
-SRC_URI += "file://0001-typefind-min-1k.patch \
-    file://0002-Small-robustness-fixes.patch"
+SRC_URI_append = "\
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0_1.16.%.bbappend:3]: Could you please add space between " and / ?

File: recipes-wpe/wpewebkit/wpewebkit_20170728.bb [hunk 19]
```diff
@@ -7,8 +7,10 @@ PR = "r2"
 
 SRCREV ?= "f9402295bed27deb780d38e33c20e397eccb009a"
 SRC_URI = "git://github.com/WebPlatformForEmbedded/WPEWebKit.git;protocol=git;branch=wpe-20170728"
-SRC_URI += "file://0001-Fix-build-with-musl.patch"
-SRC_URI += "file://0002-Define-MESA_EGL_NO_X11_HEADERS-when-not-using-GLX.patch"
+SRC_URI_append = "\
```
Comment (artur-gebicz-ml) [inline recipes-wpe/wpewebkit/wpewebkit_20170728.bb:10]: Could you please add space?

**Review by artur-gebicz-ml — CHANGES_REQUESTED** [review]

File: recipes-devtools/cmake/cmake_3.7.2.bb [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+        bb.fatal('docdir must contain prefix as its prefix')
+
+    docdir_stripped = docdir[len(prefix):]
+    if len(docdir_stripped) > 0 and docdir_stripped[0] == '/':
+        docdir_stripped = docdir_stripped[1:]
+
+    d.setVar("docdir_stripped", docdir_stripped)
+}
+
+EXTRA_OECMAKE=" \
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply recipes-devtools/cmake/cmake_3.7.2.bb:26]: done

File: recipes-devtools/cmake/cmake-native_3.7.2.bb [hunk 21]
```diff
@@ -0,0 +1,39 @@
+require cmake.inc
+inherit native
+
+DEPENDS_append = " bzip2-replacement-native expat-native xz-native zlib-native curl-native"
+
+SRC_URI_append = "\
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply recipes-devtools/cmake/cmake-native_3.7.2.bb]: done

File: recipes-devtools/cmake/cmake.inc [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+# Fixes errors like this in cmake(-native).do_configure:
+#| -- Performing Test run_pic_test - Success
+#| CMake Error at tmp-eglibc/sysroots/qemuarm/usr/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:27 (message):
+#|   The imported target "Qt5::Core" references the file
+#|
+#|      "/qmake"
+#|
+#|   but this file does not exist.  Possible reasons include:
+
+do_configure_prepend() {
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply recipes-devtools/cmake/cmake.inc:42]: done

File: recipes-devtools/cmake/cmake.inc [hunk 23]
```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2005, Koninklijke Philips Electronics NV.  All Rights Reserved
+# Released under the MIT license (see packages/COPYING)
+
+SUMMARY = "Cross-platform, open-source make system"
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply recipes-devtools/cmake/cmake.inc:4]: done

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc (hunk 15 above)
Comment (HaseenaSainul) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc:5]: done

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc (hunk 16 above)
Comment (HaseenaSainul) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc]: done

File: recipes-multimedia/gstreamer/gstreamer1.0_1.10.4.bb (hunk 17 above)
Comment (HaseenaSainul) [reply recipes-multimedia/gstreamer/gstreamer1.0_1.10.4.bb]: done

File: recipes-multimedia/gstreamer/gstreamer1.0_1.16.%.bbappend (hunk 18 above)
Comment (HaseenaSainul) [reply recipes-multimedia/gstreamer/gstreamer1.0_1.16.%.bbappend]: done

File: recipes-wpe/wpewebkit/wpewebkit_20170728.bb (hunk 19 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpewebkit/wpewebkit_20170728.bb]: done

File: recipes-graphics/libepoxy/libepoxy_1.4.0.bb [hunk 24]
```diff
@@ -1,4 +1,8 @@
 SUMMARY = "OpenGL function pointer management library"
+DESCRIPTION = "Epoxy is a library for handling OpenGL function pointer management \
```
Comment (artur-gebicz-ml) [inline recipes-graphics/libepoxy/libepoxy_1.4.0.bb:2]: Could you please indent multiline with 4 spaces?

File: recipes-graphics/openjpeg/openjpeg_2.3.1.bb [hunk 25]
```diff
@@ -3,9 +3,9 @@ HOMEPAGE = "http://www.openjpeg.org"
 LICENSE = "BSD-2-Clause"
```
Comment (artur-gebicz-ml) [inline recipes-graphics/openjpeg/openjpeg_2.3.1.bb:3]: Could you please add SUMMARY?

File: recipes-graphics/westeros/westeros.bb [hunk 26]
```diff
@@ -2,7 +2,7 @@ require westeros.inc
 
 SUMMARY = "This receipe compiles the westeros compositor component"
```
Comment (artur-gebicz-ml) [inline recipes-graphics/westeros/westeros.bb:3]: Could you please add DESCRIPTION?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc [hunk 27]
```diff
@@ -1,9 +1,11 @@
 SUMMARY = "Plugins for the GStreamer multimedia framework 1.x"
+DESCRIPTION = "GStreamer is a multimedia framework for encoding and decoding video and sound. \
+It supports a wide range of formats including mp3, ogg, avi, mpeg and quicktime."
```
Comment (artur-gebicz-ml) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins.inc:3]: Could you please indent with 4 spaces?

**Review by artur-gebicz-ml — CHANGES_REQUESTED** [review]

File: recipes-devtools/cmake/cmake-native_3.7.2.bb [hunk 28]
```diff
@@ -3,17 +3,16 @@ inherit native
...
 do_configure[cleandirs] = "${B}"
+PACKAGECONFIG[ncurser] = "-DBUILD_CursesDialog=1,-DBUILD_CursesDialog=0,ncurses-native"
 
-# Disable ccmake since we don't depend on ncurses
 CMAKE_EXTRACONF = "\
     -DCMAKE_LIBRARY_PATH=${STAGING_LIBDIR_NATIVE} \
-    -DBUILD_CursesDialog=0 \
     -DCMAKE_USE_SYSTEM_LIBRARIES=1 \
     -DCMAKE_USE_SYSTEM_LIBRARY_JSONCPP=0 \
     -DCMAKE_USE_SYSTEM_LIBRARY_LIBARCHIVE=0 \
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply recipes-devtools/cmake/cmake-native_3.7.2.bb:18]: done

File: recipes-devtools/cmake/cmake-native_3.7.2.bb [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+
+SRC_URI_append = "\
+    file://cmlibarchive-disable-ext2fs.patch \
+"
+
+B = "${WORKDIR}/build"
+do_configure[cleandirs] = "${B}"
+
+# Disable ccmake since we don't depend on ncurses
+CMAKE_EXTRACONF = "\
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply recipes-devtools/cmake/cmake-native_3.7.2.bb:14]: done

File: recipes-graphics/libepoxy/libepoxy_1.4.0.bb (hunk 24 above)
Comment (HaseenaSainul) [reply recipes-graphics/libepoxy/libepoxy_1.4.0.bb:2]: done

File: recipes-graphics/westeros/westeros.bb (hunk 26 above)
Comment (HaseenaSainul) [reply recipes-graphics/westeros/westeros.bb:1]: done

**Review by artur-gebicz-ml — APPROVED** [review]

**Pull request merged**